src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -1820,8 +1853,15 @@ private void visitEnumType(String symbolName, String qualifiedName, EnumType typ
       for (String elem : sorted(type.getElements())) {
         emit(elem);
         emit(":");
-        // No need to use type.getMembersType(), this must match the type alias we just declared.
-        emit(unqualifiedName);
+        // For string enums that have some literal values and some calculated values try to use the
+        // literal values as types as much as possible.
+        Node n = elements.get(elem);
+        if (primitiveType.equals(stringType) && n.isString()) {
+          emit("'" + n.getString() + "'");
```

Comment:
does this makes sense if primitiveType != stringType, is this valid closure

/** @enum {string|number} */
let x = {A: 0, B: "fsds"}

Reply:
This is actually a common pattern in google3 :scream: 

I don't understand your question here. `emit("'" + n.getString() + "'");` doesn't make sense if primitiveType != stringType. That's why the line above checked the type of primitiveType.

Anyways, I added a test case for this mixed enums.